Nightly Trilinos failures, all backends: error: no template named 'SerialTrsvInternalUpper' in namespace 'KokkosBatched'

**Description**:

Nightly builds of Trilinos fail to compile some packages (ifpack2, intrepid2) with output:

**Intrepid2**:
```
17:47:05 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/intrepid2/src/Projection/Intrepid2_ProjectionTools.hpp:548:11: error: no template named 'SerialTrsvInternalUpper' in namespace 'KokkosBatched'; did you mean 'KokkosBatched::Impl::SerialTrsvInternalUpper'?
17:47:05           KokkosBatched::SerialTrsvInternalUpper<KokkosBatched::Algo::Trsv::Unblocked>::invoke(false,
17:47:05           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17:47:05           KokkosBatched::Impl::SerialTrsvInternalUpper
17:47:05 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/kokkos-kernels/batched/dense/impl/KokkosBatched_Trsv_Serial_Internal.hpp:131:8: note: 'KokkosBatched::Impl::SerialTrsvInternalUpper' declared here
17:47:05 struct SerialTrsvInternalUpper {
17:47:05        ^
17:47:05 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Build/packages/intrepid2/unit-test/Projection/test_project_fields_Serial_DOUBLE.cpp:15:
17:47:05 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/intrepid2/unit-test/Projection/test_project_fields.hpp:34:
17:47:05 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/packages/intrepid2/src/Projection/Intrepid2_ProjectionTools.hpp:589:11: error: no template named 'SerialTrsvInternalUpper' in namespace 'KokkosBatched'; did you mean 'KokkosBatched::Impl::SerialTrsvInternalUpper'?
17:47:05           KokkosBatched::SerialTrsvInternalUpper<KokkosBatched::Algo::Trsv::Unblocked>::invoke(false,
17:47:05           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
17:47:05           KokkosBatched::Impl::SerialTrsvInternalUpper
17:47:05 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_LLVM1507_Serial_debug/Trilinos/kokkos-kernels/batched/dense/impl/KokkosBatched_Trsv_Serial_Internal.hpp:131:8: note: 'KokkosBatched::Impl::SerialTrsvInternalUpper' declared here
17:47:05 struct SerialTrsvInternalUpper {
17:47:05        ^
```

**Ifpack2**:
```
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp(4131): error: namespace "KokkosBatched" has no member "SerialTrsvInternalLower"
19:42:03           detected during:
19:42:03             instantiation of "void Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::operator()(const Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::SingleVectorTag<B> &, const Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::member_type &) const [with MatrixType=Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>, B=3]" 
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/kokkos/core/src/Serial/Kokkos_Serial_Parallel_Team.hpp(238): here
19:42:03             instantiation of "std::enable_if_t<<expression>, void> Kokkos::Impl::ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>, Kokkos::Serial>::exec<TagType>(Kokkos::Impl::HostThreadTeamData &) const [with FunctorType=Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>, Properties=<Kokkos::HostSpace::execution_space, Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>::SingleVectorTag<3>>, TagType=Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>::SingleVectorTag<3>]" 
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/kokkos/core/src/Serial/Kokkos_Serial_Parallel_Team.hpp(266): here
19:42:03             instantiation of "void Kokkos::Impl::ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>, Kokkos::Serial>::execute() const [with FunctorType=Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>, Properties=<Kokkos::HostSpace::execution_space, Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>::SingleVectorTag<3>>]" 
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/kokkos/core/src/Kokkos_Parallel.hpp(146): here
19:42:03             instantiation of "void Kokkos::parallel_for(const std::__cxx11::string &, const ExecPolicy &, const FunctorType &) [with ExecPolicy=Kokkos::TeamPolicy<Kokkos::HostSpace::execution_space, Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>::SingleVectorTag<3>>, FunctorType=Ifpack2::BlockTriDiContainerDetails::SolveTridiags<Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>>, Enable=void]" 
19:42:03 (4816): here
19:42:03             instantiation of "void Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::run(const Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::impl_scalar_type_2d_view_tpetra &, const Ifpack2::BlockTriDiContainerDetails::SolveTridiags<MatrixType>::impl_scalar_type_1d_view &) [with MatrixType=Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>]" 
19:42:03 (5007): here
19:42:03             instantiation of "int Ifpack2::BlockTriDiContainerDetails::applyInverseJacobi(const Teuchos::RCP<const Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_row_matrix_type> &, const Teuchos::RCP<const Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_crs_graph_type> &, const Teuchos::RCP<const Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_import_type> &, const Teuchos::RCP<Ifpack2::BlockTriDiContainerDetails::AsyncableImport<MatrixType>> &, __nv_bool, const Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_multivector_type &, Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_multivector_type &, Ifpack2::BlockHelperDetails::ImplType<MatrixType>::tpetra_multivector_type &, Ifpack2::BlockHelperDetails::ImplType<MatrixType>::impl_scalar_type_1d_view &, const Ifpack2::BlockHelperDetails::PartInterface<MatrixType> &, const Ifpack2::BlockTriDiContainerDetails::BlockTridiags<MatrixType> &, const Ifpack2::BlockHelperDetails::AmD<MatrixType> &, Ifpack2::BlockHelperDetails::ImplType<MatrixType>::vector_type_1d_view &, Ifpack2::BlockHelperDetails::NormManager<MatrixType> &, const Ifpack2::BlockHelperDetails::ImplType<MatrixType>::impl_scalar_type &, __nv_bool, int, Ifpack2::BlockHelperDetails::ImplType<MatrixType>::magnitude_type, int) [with MatrixType=Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>]" 
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp(317): here
19:42:03             instantiation of "void Ifpack2::BlockTriDiContainer<MatrixType, Ifpack2::BlockTriDiContainerDetails::ImplSimdTag>::applyInverseJacobi(const Ifpack2::BlockTriDiContainer<MatrixType, Ifpack2::BlockTriDiContainerDetails::ImplSimdTag>::mv_type &, Ifpack2::BlockTriDiContainer<MatrixType, Ifpack2::BlockTriDiContainerDetails::ImplSimdTag>::mv_type &, Ifpack2::BlockTriDiContainer<MatrixType, Ifpack2::BlockTriDiContainerDetails::ImplSimdTag>::scalar_type, __nv_bool, int) const [with MatrixType=Tpetra::RowMatrix<double, int, longlong, Tpetra_KokkosCompat_KokkosSerialWrapperNode>]" 
19:42:03 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Build/packages/ifpack2/src/Ifpack2_BlockTriDiContainer_DOUBLE_INT_LONG_LONG_SERIAL.cpp(24): here
...
```

These errors began following merge of #2452 .

Trilinos builds enable deprecated code, I'm not sure if that indicates an inadvertent compatibility change that should be addressed, or if changes should be made in Trilinos itself (which will be necessary regardless, I suppose)


**Reproducer** (blake all queue):
```bash

# Repo prep
git clone -b develop https://github.com/trilinos/Trilinos.git
TRILINOS_DIR=$PWD/Trilinos
git clone -b develop https://github.com/kokkos/kokkos.git
KOKKOS_DIR=$PWD/kokkos
git clone -b develop https://github.com/kokkos/kokkos-kernels.git
KOKKOSKERNELS_DIR=$PWD/kokkos-kernels

cd $TRILINOS_DIR
ln -s ${KOKKOS_DIR}/kokkos kokkos
ln -s ${KOKKOSKERNELS_DIR}/kokkos-kernels kokkos-kernels
cd -

mkdir -p Build
cd Build

salloc -N 1 -p all

# Environment
module purge
module load cmake llvm openblas/0.3.23
module list

# Configure Trilinos
cmake \
  -D CMAKE_INSTALL_PREFIX="${TRILINOS_INSTALL_DIR}" \
  -D CMAKE_CXX_COMPILER="`which clang++`" \
  -D CMAKE_C_COMPILER="`which clang`" \
  -D CMAKE_Fortran_COMPILER="`which gfortran`" \
  -D CMAKE_CXX_FLAGS="-g -fPIC" \
  -D CMAKE_C_FLAGS="-g -fPIC" \
  -D CMAKE_Fortran_FLAGS="-g -fPIC" \
  -D CMAKE_BUILD_TYPE:STRING=DEBUG \
  -D BUILD_SHARED_LIBS:BOOL=OFF \
  -DTPL_ENABLE_MPI=OFF \
  -DTPL_ENABLE_BLAS:BOOL=ON \
   -DBLAS_LIBRARY_DIRS:FILEPATH=${OPENBLAS_ROOT}/lib \
   -DBLAS_LIBRARY_NAMES:STRING="openblas" \
  -DTPL_ENABLE_LAPACK:BOOL=ON \
   -DLAPACK_INCLUDE_DIRS:FILEPATH=${OPENBLAS_ROOT}/include \
   -DLAPACK_LIBRARY_DIRS:FILEPATH=${OPENBLAS_ROOT}/lib \
   -DLAPACK_LIBRARY_NAMES:STRING="openblas" \
  -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
  -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_MUST_FIND_ALL_TPL_LIBS=TRUE \
  -DTrilinos_ENABLE_COMPLEX=ON \
  -DTrilinos_ENABLE_OpenMP=OFF \
  -DTrilinos_ENABLE_Kokkos=ON \
  -D Kokkos_ENABLE_SERIAL=ON \
   -D Kokkos_ENABLE_TESTS=ON \
  -D Kokkos_ARCH_SKX=ON \
   -D Kokkos_ENABLE_IMPL_MDSPAN=ON \
  -DTrilinos_ENABLE_KokkosKernels=ON \
   -D KokkosKernels_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Tpetra=ON \
   -D Tpetra_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Sacado=ON \
   -D Sacado_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Stokhos=ON \
   -D Stokhos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Intrepid2=ON \
   -D Intrepid2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Adelus=ON \
   -D Adelus_ENABLE_TESTS:BOOL=ON \
  -DTrilinos_ENABLE_Compadre=ON \
   -D Compadre_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Panzer=ON \
   -D Panzer_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Amesos2=ON \
   -D Amesos2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Zoltan2=ON \
   -D Zoltan2_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Teuchos=ON \
   -D Teuchos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_Ifpack2=ON \
   -D Ifpack2_ENABLE_TESTS=ON \
\
  -DTPL_ENABLE_Matio=OFF \
\
-DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
-DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
\
$TRILINOS_DIR
```

--- a/batched/dense/src/KokkosBatched_Trsv_Decl.hpp
+++ b/batched/dense/src/KokkosBatched_Trsv_Decl.hpp
@@ -76,11 +76,11 @@ struct Trsv {
   KOKKOS_INLINE_FUNCTION static int invoke(const MemberType &member, const ScalarType alpha, const AViewType &A,
                                            const bViewType &b) {
     int r_val = 0;
-    if (std::is_same<ArgMode, Mode::Serial>::value) {
+    if (std::is_same_v<ArgMode, Mode::Serial>) {
       r_val = SerialTrsv<ArgUplo, ArgTrans, ArgDiag, ArgAlgo>::invoke(alpha, A, b);
-    } else if (std::is_same<ArgMode, Mode::Team>::value) {
+    } else if (std::is_same_v<ArgMode, Mode::Team>) {
       r_val = TeamTrsv<MemberType, ArgUplo, ArgTrans, ArgDiag, ArgAlgo>::invoke(member, alpha, A, b);
-    } else if (std::is_same<ArgMode, Mode::TeamVector>::value) {
+    } else if (std::is_same_v<ArgMode, Mode::TeamVector>) {
       r_val = TeamVectorTrsv<MemberType, ArgUplo, ArgTrans, ArgDiag, ArgAlgo>::invoke(member, alpha, A, b);
     }
     return r_val;
@@ -93,17 +93,29 @@ struct Trsv {
 #include "KokkosBatched_Trsv_Team_Impl.hpp"
 #include "KokkosBatched_Trsv_TeamVector_Impl.hpp"
 
-#define KOKKOSBATCHED_SERIAL_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
-  KokkosBatched::SerialTrsvInternalLower<ALGOTYPE>::invoke(DIAG::use_unit_diag, M, ALPHA, A, AS0, AS1, B, BS)
+#define KOKKOSBATCHED_SERIAL_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS)  \
+  KokkosBatched::Impl::SerialTrsvInternalLower<ALGOTYPE>::invoke(DIAG::use_unit_diag, false, M, ALPHA, A, AS0, AS1, B, \
+                                                                 BS)
 
-#define KOKKOSBATCHED_SERIAL_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
-  KokkosBatched::SerialTrsvInternalUpper<ALGOTYPE>::invoke(DIAG::use_unit_diag, N, ALPHA, A, AS1, AS0, B, BS)
+#define KOKKOSBATCHED_SERIAL_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS)     \
+  KokkosBatched::Impl::SerialTrsvInternalUpper<ALGOTYPE>::invoke(DIAG::use_unit_diag, false, N, ALPHA, A, AS1, AS0, B, \
+                                                                 BS)
 
-#define KOKKOSBATCHED_SERIAL_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
-  KokkosBatched::SerialTrsvInternalUpper<ALGOTYPE>::invoke(DIAG::use_unit_diag, M, ALPHA, A, AS0, AS1, B, BS)
+#define KOKKOSBATCHED_SERIAL_TRSV_LOWER_CONJTRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
+  KokkosBatched::Impl::SerialTrsvInternalUpper<ALGOTYPE>::invoke(DIAG::use_unit_diag, true, N, ALPHA, A, AS1, AS0, B,  \
+                                                                 BS)
 
-#define KOKKOSBATCHED_SERIAL_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
-  KokkosBatched::SerialTrsvInternalLower<ALGOTYPE>::invoke(DIAG::use_unit_diag, N, ALPHA, A, AS1, AS0, B, BS)
+#define KOKKOSBATCHED_SERIAL_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS)  \
+  KokkosBatched::Impl::SerialTrsvInternalUpper<ALGOTYPE>::invoke(DIAG::use_unit_diag, false, M, ALPHA, A, AS0, AS1, B, \
+                                                                 BS)
+
+#define KOKKOSBATCHED_SERIAL_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS)     \
+  KokkosBatched::Impl::SerialTrsvInternalLower<ALGOTYPE>::invoke(DIAG::use_unit_diag, false, N, ALPHA, A, AS1, AS0, B, \
+                                                                 BS)
+
+#define KOKKOSBATCHED_SERIAL_TRSV_UPPER_CONJTRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS) \
+  KokkosBatched::Impl::SerialTrsvInternalLower<ALGOTYPE>::invoke(DIAG::use_unit_diag, true, N, ALPHA, A, AS1, AS0, B,  \
+                                                                 BS)
 
 #define KOKKOSBATCHED_TEAM_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, \
                                                                    B, BS)                                            \
@@ -143,46 +155,46 @@ struct Trsv {
 
 #define KOKKOSBATCHED_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(MODETYPE, ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0,   \
                                                               AS1, B, BS)                                              \
-  if (std::is_same<MODETYPE, KokkosBatched::Mode::Serial>::value) {                                                    \
+  if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Serial>) {                                                         \
     KOKKOSBATCHED_SERIAL_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);     \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::Team>::value) {                                               \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Team>) {                                                    \
     KOKKOSBATCHED_TEAM_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B,    \
                                                                BS);                                                    \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::TeamVector>::value) {                                         \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::TeamVector>) {                                              \
     KOKKOSBATCHED_TEAMVECTOR_TRSV_LOWER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, \
                                                                      B, BS);                                           \
   }
 
 #define KOKKOSBATCHED_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(MODETYPE, ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, \
                                                            B, BS)                                                      \
-  if (std::is_same<MODETYPE, KokkosBatched::Mode::Serial>::value) {                                                    \
+  if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Serial>) {                                                         \
     KOKKOSBATCHED_SERIAL_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);        \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::Team>::value) {                                               \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Team>) {                                                    \
     KOKKOSBATCHED_TEAM_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);  \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::TeamVector>::value) {                                         \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::TeamVector>) {                                              \
     KOKKOSBATCHED_TEAMVECTOR_TRSV_LOWER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B, \
                                                                   BS);                                                 \
   }
 
 #define KOKKOSBATCHED_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(MODETYPE, ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0,   \
                                                               AS1, B, BS)                                              \
-  if (std::is_same<MODETYPE, KokkosBatched::Mode::Serial>::value) {                                                    \
+  if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Serial>) {                                                         \
     KOKKOSBATCHED_SERIAL_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);     \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::Team>::value) {                                               \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Team>) {                                                    \
     KOKKOSBATCHED_TEAM_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B,    \
                                                                BS);                                                    \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::TeamVector>::value) {                                         \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::TeamVector>) {                                              \
     KOKKOSBATCHED_TEAMVECTOR_TRSV_UPPER_NO_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, \
                                                                      B, BS);                                           \
   }
 
 #define KOKKOSBATCHED_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(MODETYPE, ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, \
                                                            B, BS)                                                      \
-  if (std::is_same<MODETYPE, KokkosBatched::Mode::Serial>::value) {                                                    \
+  if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Serial>) {                                                         \
     KOKKOSBATCHED_SERIAL_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);        \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::Team>::value) {                                               \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::Team>) {                                                    \
     KOKKOSBATCHED_TEAM_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B, BS);  \
-  } else if (std::is_same<MODETYPE, KokkosBatched::Mode::TeamVector>::value) {                                         \
+  } else if (std::is_same_v<MODETYPE, KokkosBatched::Mode::TeamVector>) {                                              \
     KOKKOSBATCHED_TEAMVECTOR_TRSV_UPPER_TRANSPOSE_INTERNAL_INVOKE(ALGOTYPE, MEMBER, DIAG, M, N, ALPHA, A, AS0, AS1, B, \
                                                                   BS);                                                 \
   }

